dali/operators/reader/loader/webdataset_loader.h

```diff
@@ -32,8 +31,37 @@ namespace dali {
 namespace detail {
 namespace wds {
 
-const std::string kCurrentIndexVersion = "v1.2";  // NOLINT
-const std::unordered_set<std::string> kSupportedIndexVersions = {"v1.1", kCurrentIndexVersion};
+constexpr int ParseIndexVersion(const char* index_version) {
+  const char* it = nullptr;
+  for (it = index_version; *it != '\0'; it++) {}
+  int mult = 1, ret = 0;
+  auto update_ret = [&](auto it) {
+    assert(*it >= '0' && *it <= '9');
+    ret += mult * (*it - '0');
+  };
+  assert(*it == '\0');
+  update_ret(--it);
+  mult *= 10;
+  if (*--it != '.') {
+    update_ret(it--);
+  }
+  assert(*it == '.');
+  mult *= 10;
+  while (*--it != 'v') {
+    mult *= 10;
+    update_ret(it);
+  }
+  return ret;
+}
+
+bool DLL_PUBLIC VerifyIndexVersionString(const std::string& index_version_string);
+
+// Minor part of the index version may be at most two-digit.
+const char kCurrentIndexVersion[] = "v1.2";
```

Comment:
How about:
```
enum wbds_idx_version{
  WBS_IDX_V1_1,
  WBS_IDX_V1_2
};

const vector<int> WbsIdxVersions = {
  ParseIndexVersion("v1.2"),
  ParseIndexVersion("v1.2")
};

const std::set<int> kSupportedIndexVersions(WbsIdxVersions.begin(), WbsIdxVersions.end());

bool CompareWbsIdxVersion(int in, wbds_idx_version ref) {
  return WbsIdxVersions[ref] == in;
}
```
so you can cheaply
```
CompareWbsIdxVersion(index_version, WBS_IDX_V1_2);
```
I don't like the idea of parsing version string over and over. That is why we wants int representation.

Reply:
I think we should have a well defined way to convert a (major, minor) into a single numerical value, e.g.:
```C++
constexpr int MakeWDSIndexVersion(int major, int minor) {
  return major * 1000 + minor;
}
```
Enum values are therefore not really necessary. We could have sth like:
```C++
const std::set<int> supported_index_versions  = {
  MakeWDSIndexVersion(1, 1),
  MakeWDSIndexVersion(1, 2),
};

constexpr int kCurrentWDSIndexVersion = MakeWDSIndexVersion(1, 2);
```